### PR TITLE
Update policies when entities referenced by name change their names. …

### PR DESCRIPTION
**Pull request by plorenz**
…Fixes GH-43

### CODE REVIEW

File: controller/persistence/base_store.go [hunk 1]
```diff
@@ -108,6 +108,48 @@ func (store *baseStore) baseLoadOneById(tx *bbolt.Tx, id string, entity boltz.Ba
 	return nil
 }
 
+func (store *baseStore) updateEntityNameReferences(bucket *boltz.TypedBucket, rolesSymbol boltz.EntitySetSymbol, entity NamedEdgeEntity, oldName string) {
+	// If the entity name is the same as entity ID, bail out. We don't want to remove any references by ID, since
+	// those take precedence over named references
+	if store.IsEntityPresent(bucket.Tx(), oldName) {
+		return
```
Comment (dovholuknf) [inline controller/persistence/base_store.go]: i feel like this needs a log, or to return an err maybe? If not - why not?

File: controller/persistence/base_store.go [hunk 2]
```diff
@@ -108,6 +108,48 @@ func (store *baseStore) baseLoadOneById(tx *bbolt.Tx, id string, entity boltz.Ba
...
+		})
+		if err != nil {
+			bucket.SetError(err)
+			return
+		}
+	}
+}
+
+func (store *baseStore) deleteEntityReferences(tx *bbolt.Tx, entity NamedEdgeEntity, rolesSymbol boltz.EntitySetSymbol) error {
+	// If the entity name is the same as entity ID, don't remove any of those references as id references take precedence
```
Comment (dovholuknf) [inline controller/persistence/base_store.go]: same comment as above. seems like a really hard situation to detect in the off chance this happens?

File: controller/persistence/base_store.go (hunk 1 above)
Comment (plorenz) [reply controller/persistence/base_store.go]: I'll add a log statement

File: controller/persistence/base_store.go (hunk 2 above)
Comment (plorenz) [reply controller/persistence/base_store.go]: I'll add a log statement

**Pull request merged**